### PR TITLE
Feature: Adding CRC32 checksums to TSM Page Files

### PR DESCRIPTION
**Pull request by liikeuforever**
# Required checklist
- [ ] Sample config files updated (`config`,`meta/config` and `default config`) 
- [ ] If there are user-facing changes, the documentation needs to be updated prior to approving the PR( [Link]() )
- [ ] If there are any breaking changes to public APIs, please add the `api change` label.
- [ ] Signed [CLA](https://cla-assistant.io/cnosdb/cnosdb) (if not already signed)

# Which issue does this PR close?

[//]: # (We generally require a GitHub issue to be filed for all bug fixes and enhancements and this helps us generate change logs for our releases. You can link an issue to this PR using the GitHub syntax. For -- example `Closes #123` indicates that this PR will close issue #123.)

Related #.

# Rationale for this change

[//]: # (Why are you proposing this change? If this is already explained clearly in the issue then this section is not needed.
 Explaining clearly why changes are proposed helps reviewers understand your changes and offer better suggestions for fixes.)

# Are there any user-facing changes?

[//]: # (There is no need to duplicate the description in the issue here, but it is sometimes worth providing a summary of the individual changes in this PR.)

### CODE REVIEW

File: tskv/src/tsm/reader.rs [hunk 1]
```diff
@@ -307,7 +307,7 @@ pub fn decode_buf_to_pages(
             meta: page.meta.clone(),
             bytes: Bytes::from(page_buf.to_vec()),
         };
-        pages.push(page);
+        pages.push(page.crc_validation().unwrap());
```
Comment (Subsegment) [inline tskv/src/tsm/reader.rs:310]: return result

File: tskv/src/tsm/writer.rs [hunk 2]
```diff
@@ -385,9 +385,20 @@ impl Column {
...
-        data.extend_from_slice(self.valid.bytes());
+        let mut hasher = crc32fast::Hasher::new();
+        let len_bitset = len_bitset.to_be_bytes();
+        hasher.update(&len_bitset);
+        let data_len = data_len.to_be_bytes();
+        hasher.update(&data_len);
+        let binding = self.valid.clone();
+        let valid = binding.bytes();
+        hasher.update(&valid);
+        hasher.update(&buf);
```
Comment (Subsegment) [inline tskv/src/tsm/writer.rs:396]: only crchash buf

Comment (liikeuforever) [reply tskv/src/tsm/writer.rs:393]: done

File: tskv/src/tsm/reader.rs (hunk 1 above)
Comment (liikeuforever) [reply tskv/src/tsm/reader.rs]: done

File: tskv/src/tsm/reader.rs [hunk 3]
```diff
@@ -307,7 +307,15 @@ pub fn decode_buf_to_pages(
             meta: page.meta.clone(),
             bytes: Bytes::from(page_buf.to_vec()),
         };
-        pages.push(page);
+        let page_result = page.crc_validation();
```
Comment (Subsegment) [inline tskv/src/tsm/reader.rs:310]: use **page.crc_validation()?;**

File: tskv/src/tsm/writer.rs [hunk 4]
```diff
@@ -385,8 +385,16 @@ impl Column {
...
         };
         let mut data = vec![];
-        data.extend_from_slice(&len_bitset.to_be_bytes());
-        data.extend_from_slice(&data_len.to_be_bytes());
+        let mut hasher = crc32fast::Hasher::new();
+        let len_bitset = len_bitset.to_be_bytes();
+        hasher.update(&len_bitset);
+        let data_len = data_len.to_be_bytes();
+        hasher.update(&data_len);
+        hasher.update(&buf);
```
Comment (Subsegment) [inline tskv/src/tsm/writer.rs:393]: only hasher.update(&buf)

File: tskv/src/tsm/page.rs [hunk 5]
```diff
@@ -48,6 +49,26 @@ impl Page {
         &self.meta.column
     }
 
+    pub fn crc_validation(&self) -> Result<Page> {
```
Comment (Subsegment) [inline tskv/src/tsm/page.rs:52]: add unit test case

**Review by Subsegment — APPROVED** [review]

**Pull request merged**